BigLab.py

```diff
@@ -0,0 +1,305 @@
+        if(pair[1] < min_d[1]):
+            min_d = pair
+            
+    return min_d
+
+# Данный метод необходим для нахождения самой правой точки ЛЕВОЙ ножки стола
+
+def find_left_leg(contour, dot):
+    dot[0]+=50;
+    print(dot)
+    dots = []
+    
+    for i in range(len(contour)):
+        pair = contour[i]
+        if((pair[0] > dot[0] + 50) and pair[1] < dot[1]):
+            dots.append(pair)
+            
+    min_d = x_min2    
+    
+    for i in range(len(dots)):
+        pair = dots[i]
+        if(pair[1] > min_d[1]):
+            min_d = pair
+            
+    return min_d
+
+# Данный мето нужен, чтобы опрелелить точку, отвечающую за нижнюю точку столешницы стола
+
+def find_dot(contour):
+    for i in range(len(contour)):
+        pair = contour[i]
+        if(pair[1] == x_med[1]):
+            print(pair)
+            return pair
+
+        # Метод для нахождения левой, правой, верхней и нижней точки, а также для нахождения ширины и высоты объектов
+def find_square(contour):
+    max_x = contour[0]
+    max_y = contour[0]
+    min_x = contour[0]
+    min_y= contour[0]
+    for i in range(len(contour)):
+        pair = contour[i]
+        if max_x[1] < pair[1]:
+            max_x = pair
+        if min_x[1] > pair[1]:
+            min_x = pair
+        if min_y[0] < pair[0]:
+            min_y = pair
+        if max_y[0] > pair[0]:
+            max_y = pair
+    
+    
+    if(max_y[1] < min_x[1]):
+        min_x[1] = max_y[1]
+        
+    
+    width = max_x[1] - min_x[1] 
+    height = min_y[0] - max_y[0]
+    
+    return round(width * height), max_x, min_x, max_y, min_y, width, height
+
+# Загрузим фотографию и удем искать наши объекты с помощью методов определения границ по цвету
+# Так как стул обернут в синюю ткань, а внешний вид стола был подвергнут маодернизации
+# ввиду проблем с нахождением границ такими методами как Canny или Хаффа
+# Мы будем искать на изображении оттенки синего в диапазоне этого цвета
+
+image = cv2.imread('./dataset/IMG_9844.jpg')
+fig, ax = plt.subplots(1, 3, figsize=(15, 6))
+
+image_2 = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+
+hsv_image = cv2.cvtColor(image_2, cv2.COLOR_RGB2HSV)
+light_blue = (82, 25, 0)                                            # Нижняя граница синего цвета в палитре HSV
+dark_blue = (161, 255, 255)                                         # Верхняя граница синего цвета в палитре HSV
+mask_blue = cv2.inRange(hsv_image, light_blue, dark_blue)
+result_blue = cv2.bitwise_and(image, image_2, mask=mask_blue)
+
+ax[0].imshow(image)
+ax[1].imshow(image_2)
+ax[2].imshow(hsv_image)
+fig1, ax1 = plt.subplots(1, 2, figsize=(15, 6))
+ax1[0].imshow(mask_blue, cmap="gray")
+ax1[1].imshow(result_blue)
+isWritten = cv2.imwrite('./dataset/work_image.jpg', mask_blue)
+
+# Полученное изображение контуров белого цвета на черном фоне пропускаем через Canny и находим границы наших объектов
+
+binary_img = binary_closing(canny(mask_blue, sigma=0.8), selem=np.ones((10, 10))) #1;6

```

Comment:
Нет смысла применять фильтр Canny к бинаризованной маске, для нахождения границ есть болле оптимальные методы (например find_contours, который далее применяешь). Так как у тебя получается несколько областей связности, советую взять selem=np.ones((10, 10)) побольше.
```suggestion
binary_img = binary_closing(mask_blue, selem=np.ones((30, 30)))#1;6

```

Reply:
Советую также попробовать использовать последовательность делитации и эрозии с примитивами разных размеров. Для делитации ~(15, 15), для эрозии ~(40, 40)

Reply:
возможно стоит попробовать искать области связанности отдельно, например как было рассмотрено во 2 мини лабе

Reply:
Спасибо за наводку, попробую, возможно залью с этими исправлениями